Roles to start uptane and tear it down, on demo server

d80ep08th opened the pull request:
Added Roles: 
- install-prereqs : Ensure all the tools required for uptane exist on the server
- configure-demo-server: Ensure the server is configured to be able to start uptane
- start-server: Ensure that uptane up and running
- soft-reset: Ensure that uptane is down and all keys are absent

tkfu reviewed (changes requested):
There's a bit of a start here, but needs a lot of revisions before it can be merged. In addition to the specific issues referenced in the review, there's no error handling. Remember, this will run as an automated process; you need to catch errors and deal with them appropriately.

tkfu reviewed (changes requested):
There's some good progress here, but still a long way to go. There's basically nothing in the way of error handling.

d80ep08th committed:
Created new roles for ota-ce on the demo server

- install-prereqs
This role should be run just after the server has been wiped, but can be run at any
subsequent time. It makes sure that the necessary packages for running ota-ce are installed
on the demo server.

- configure-demo-server
This role should only be run after the server has been wiped and all the necessary
packages are installed on the demo-server. It will throw error if you run this role
after this role has been already run once. It configures the ota-ce to  be capable
of starting all its services on the demo-server and setup the landing page for the server.

- start-server
This role should be run after the demo server is configured and after a soft-wipe.
It will start all the services in docker containers and create key material, making
sure the services are online and available to use for the internet.

- soft-wipe-demo-server
This role should be run after the ota-ce services have been started.
This role will tear down all services and delete all key material and user data.

d80ep08th committed:
Created new playbooks

- hard-reset.yml
This playbook will reset the server completely and can be run anytime.
It starts by wiping out everyting on the server, then bringing up the firewall
and then installing all the necessary software to run ota-ce. Then it configures
ota-ce and runs the services.

- setup-server.yml
This playbook will wipe reset the server completely and then configure ota-ce on it.
It does exactly what hard-reset playbook does except it will not start the services
on the server. It will make sure that ota-ce is configured to run on the server.

- soft-reset.yml
This playbook will reset the server and start it again without wiping out the whole server.
It will bring down all the services and delete all key material, then it will start up those
services again.

- demo-uptane.yml
This playbook can be run once the server is completely wiped out and the firewall is up.
will install all the prerequisite software, configure the server to run ota-ce
and then start the services on the server. If all services are down and only the uptane
directory is deleted from the server, this playbook can be run to set uptane back up again.

d80ep08th committed:
Created landing page for the demo server

- landing-page.conf
Contains the nginx configuration to display landing page on uptanedemo.org

- index.html
Landing page's index file that will display the html content.

- landing-page-framework.zip
Contains the necessary js, css and fonts files to add to the user experience of the landing page.

All these files are copied over to the landing-page's nginx container while playing the
configure-demo-server role

d80ep08th committed:
Refactored documentation
- added changes in the roles for hard-reset.yml
- changed name of the configure-security role to start-firewall
- removed the role to create credentials
- added deescription of demo-uptane playbook's roles
